.github/workflows/ci.yml

```diff
@@ -0,0 +1,75 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+
+env:
+  HF_SCRIPTS_VERSION: main
+  HF_ALLOW_CODE_EVAL: 1
+
+jobs:
+
+  check_code_quality:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.6"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .[quality]
+      - name: Check quality
+        run: |
+          black --check --line-length 119 --target-version py36 tests src benchmarks datasets metrics
+          isort --check-only tests src benchmarks datasets metrics
+          flake8 tests src benchmarks datasets metrics
+
+  test:
+    needs: check_code_quality
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        pyarrow_version: [latest, 6.0.1]
```

Comment:
The strategy (cross product matrix of OS and PyArrow versions) defaults to True `fail-fast`: GitHub will cancel all in-progress and queued jobs in the matrix if any job in the matrix fails.

This saves CI running time (billing).

Reply:
If the Windows tests fails, a fix is necessary.

And once fixed, the CI will be triggered again. Eventually ALL tests will be run.

The CI will only be green if and only if ALL test jobs run and pass.

Reply:
@lhoestq 
Another typical example, besides the one above about the contributor forgetting passing locally the code quality check:
- Usually, if there is a test not passing on Linux, it will not pass on Windows either (nearly for sure)
   - Therefore, if the tests are not passing on ubunut-latest, what is the point of continue running the tests on windows-latest?
      - Note Windows takes 3x more time to run than Linux and costs twice the price per minute than Linux

Reply:
Hmm actually it happens so often to have a CI that fails (because of flakiness or because of dataset card content missing) that I wouldn't stop all the jobs

Reply:
I'm not against trying this CI for a week or so, but if it's annoying we'll have to disable the CI matrix strategy

Reply:
For sure, our golden target should be to have a CI so that we only merge a PR if and only if the CI is green. I'm strongly against having a CI that we can ignore sometimes and merge even if it is red.

Reply:
Well this is currently the case and having the matrix strategy is going to be annoying I think. I personnally don't want to lose time relaunching CI jobs just because of some random error from the Hub.